src/app/app.ts

```diff
@@ -0,0 +1,39 @@
+import {Post} from 'mattermost-redux/types/posts';
+import {AppCall, AppContext} from 'mattermost-redux/types/apps';
+
+import {ENV} from '../utils';
+
+import {mattermost, zendesk} from '../clients';
+
+import store from '../store/config';
+
+import {getTicketForPost} from './model';
+
+class App {
+    createTicketFromPost = async (call: AppCall): Promise<string> => {
+        const ticket = getTicketForPost(call.values);
+
+        const zdClient = zendesk.newClient(ENV.zendesk.username, ENV.zendesk.apiToken, ENV.zendesk.apiURL);
```

Comment:
Can we have the `App` class accept these env vars in a constructor, instead of accessing `utils.ENV` directly? This will make writing tests easier. Anything that relies on the env vars should receive them in a constructor or as a function argument if it is a flat function 2/5.

Reply:
If you don't mind, I will address this in the oauth PR https://github.com/mattermost/mattermost-app-zendesk/pull/12.  I'll also create as a github issue.